src/android/com/microsoftopentechnologies/intellij/wizards/createvm/SelectImageStep.java

```diff
@@ -0,0 +1,312 @@
+        imageInfoPanel = new JPanel() {
+            @Override
+            public Dimension getPreferredSize() {
+
+                double height = 0;
+                for (Component component : this.getComponents()) {
+                    height += component.getHeight();
+                }
+
+                Dimension preferredSize = super.getPreferredSize();
+                preferredSize.setSize(preferredSize.getWidth(), height);
+                return preferredSize;
+            }
+        };
+    }
+
+
+    private enum PublicImages {
+        WindowsServer,
+        SharePoint,
+        BizTalkServer,
+        SQLServer,
+        VisualStudio,
+        Other;
+
+        @Override
+        public String toString() {
+            switch(this) {
+                case WindowsServer:
+                    return "Windows Server";
+                case BizTalkServer:
+                    return "BizTalk Server";
+                case SQLServer:
+                    return "SQL Server";
+                case VisualStudio:
+                    return "Visual Studio";
+                default:
+                    return super.toString();
+            }
+        }
+    }
+
+    private enum MSDNImages {
+        BizTalkServer,
+        Dynamics,
+        VisualStudio,
+        Other;
+
+        @Override
+        public String toString() {
+            switch (this) {
+                case BizTalkServer:
+                    return "BizTalk Server";
+                case VisualStudio:
+                    return "Visual Studio";
+                default:
+                    return super.toString();
+            }
+
+        }
+    }
+
+    private enum PrivateImages {
+        VMImages;
+
+        @Override
+        public String toString() {
+            switch (this) {
+                case VMImages:
+                    return "VM Images";
+                default:
+                    return super.toString();
+            }
+        }
+    }
+
+    Map<Enum, List<VirtualMachineImage>> virtualMachineImages;
+    private Project project;
+
+    public SelectImageStep(final CreateVMWizardModel model, Project project) {
+
+        super("Choose a Virtual Machine Image", null);
+
+        this.model = model;
+        this.project = project;
+
+        model.configStepList(createVmStepsList, 1);
+
+        final ArrayList imageTypeList = new ArrayList();
+        imageTypeList.add("Public Images");
+        imageTypeList.addAll(Arrays.asList(PublicImages.values()));
+        imageTypeList.add("MSDN Images");
+        imageTypeList.addAll(Arrays.asList(MSDNImages.values()));
+        imageTypeList.add("Private Images");
+        imageTypeList.addAll(Arrays.asList(PrivateImages.values()));
+
+
+        imageTypeComboBox.setModel(new DefaultComboBoxModel<Object>(imageTypeList.toArray()) {
+            @Override
+            public void setSelectedItem(Object o) {
+                if (o instanceof Enum) {
+                    super.setSelectedItem(o);
```

Comment:
Yes, the thing is that if the type is not an enum (there are Strings as titles), the code will not run the super code, so the selection is not made.